### PR TITLE
dinterpret.d: more use of stack temporaries

### PR DESCRIPTION
**Pull request by WalterBright**

### CODE REVIEW

File: src/dmd/dinterpret.d [hunk 1]
```diff
@@ -1089,11 +1089,11 @@ public:
         if (istate.start == s)
             istate.start = null;
 
-        size_t dim = s.statements ? s.statements.dim : 0;
-        for (size_t i = 0; i < dim; i++)
+        const dim = s.statements ? s.statements.dim : 0;
+        foreach (i; 0 .. dim)
```
Comment (jacob-carlborg) [inline src/dmd/dinterpret.d]: Please use a regular foreach loop instead of a range foreach if possible.

File: src/dmd/dinterpret.d [hunk 2]
```diff
@@ -1112,11 +1112,11 @@ public:
...
```
Comment (jacob-carlborg) [inline src/dmd/dinterpret.d]: Same here.

**Review by jacob-carlborg — CHANGES_REQUESTED** [review]

File: src/dmd/dinterpret.d (hunk 1 above)
Comment (WalterBright) [reply src/dmd/dinterpret.d]: I didn't because look at the declaration for `dim`, there's a null check that wouldn't be there in the foreach.

Comment (jacob-carlborg) [reply src/dmd/dinterpret.d]: Make sense.

**Review by jacob-carlborg — APPROVED** [review]

**Pull request merged**